preload/subresource-integrity.html

```diff
@@ -309,6 +309,39 @@
         {integrity: "sha256-deaddeadbeefYHFvsYdWumweeFAw0hJDTFt9seErghA="}
       )
 
+      SRIPreloadTest(
+        true,
+        true,
+        `Same-origin ${destination} with matching digest does not reuse preload without digest.`,
+        2,
+        destination,
+        same_origin_prefix + destination + ext + `?${token()}`,
+        {},
+        {integrity: sha256}
+      )
+
+      SRIPreloadTest(
+        true,
+        true,
+        `Same-origin ${destination} with matching digest does not reuse preload with matching but weaker digest.`,
+        2,
+        destination,
+        same_origin_prefix + destination + ext + `?${token()}`,
+        {integrity: sha384},
+        {integrity: sha256},
+      )
+
+      SRIPreloadTest(
+        true,
+        true,
+        `Same-origin ${destination} with matching digest does not reuse preload with matching but stronger digest.`,
+        2,
+        destination,
+        same_origin_prefix + destination + ext + `?${token()}`,
+        {integrity: sha256},
+        {integrity: sha384},
+      )
+
```

Comment:
How about also adding a test where the preload entry's integrity metadata is empty and the request's integrity metadata is e.g. a wrong sha-256? This is about the second case in https://github.com/whatwg/html/pull/7738:

```pre
<li><p><var>entry</var>'s <span data-x="preload integrity metadata">integrity metadata</span>
     is the empty string</p></li>
```

Reply:
Yes, that one fails in both Gecko and Chromium. That line in the spec should actually be removed - the test in line 312 also tests for this with a good digest. I'll update the spec and the sheet